File: services/nuldoc/nuldoc-src/components/StaticStylesheet.ts
import { join } from "@std/path";
import { Config } from "../config.ts";
import { Element, link } from "../dom.ts";
import { calculateFileHash } from "./utils.ts";

export default async function StaticStylesheet(
  { site, fileName, config }: {
    site?: string;
    fileName: string;
    config: Config;
  },
): Promise<Element> {
  const filePath = join(
    Deno.cwd(),
    config.locations.staticDir,
    site || "_all",
    fileName,
  );
  const hash = await calculateFileHash(filePath);
  return link({ rel: "stylesheet", href: `${fileName}?h=${hash}` });
}

File: services/nuldoc/lib/nuldoc/components/static_stylesheet.rb
module Nuldoc
  module Components
    class StaticStylesheet < DOM::HTMLBuilder
      def initialize(file_name:, config:, site: nil)
        super()
        @file_name = file_name
        @config = config
        @site = site
      end

      def build
        file_path = File.join(Dir.pwd, @config.locations.static_dir, @site || '_all', @file_name)
        hash = ComponentUtils.calculate_file_hash(file_path)
        link rel: 'stylesheet', href: "#{@file_name}?h=#{hash}"
      end
    end
  end
end
